nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/util/DataTypeUtils.java

```diff
@@ -339,6 +339,11 @@ public static boolean isRecordTypeCompatible(final Object value) {
             return dest;
         }
 
+        if (value instanceof ArrayList) {
```

Comment:
I agree with @ottobackwards. It really should not be a specific implementation.

Reply:
Great idea. I have updated.